# Resources
A persistent resources list.

## Contents

- [Bioinformatics](#bioinformatics)
  - [Bioinformatics Databases](#bioinformatics-databases)
  - [General Resources](#general-resources)
  - [Programs](#programs)
- [R](#r)
  - [R Packages](#r-packages)
- [Web Apps](#web-apps)
- [Courses](#courses)

## Bioinformatics

- [BioContainers/containers](https://github.com/BioContainers/containers) - Repository of approved bioinformatics containers
- [Code Ocean | Discover & Run Scientific Code](https://codeocean.com/) - Code Ocean is a research collaboration platform.
- [Connectivity Map sign-in](https://portals.broadinstitute.org/cmap/)
- [GSEA](http://software.broadinstitute.org/gsea/index.jsp) - Gene Set Enrichment Analysis.
- [HCA Data Portal](https://data.humancellatlas.org/) - Human Cell Atlas Data Portal. The HCA Data Portal stores and provides single-cell data contributed by labs around the world. Anyone can contribute data, find data, or access community tools and applications.
- [Broad GDAC Firehose](https://gdac.broadinstitute.org/) - 
- [Awesome-Bioinformatics](https://github.com/danielecook/Awesome-Bioinformatics) - A curated list of awesome Bioinformatics software, resources, and libraries. Mostly command line based, and free or open-source.
- [Cell Model Passports](https://cellmodelpassports.sanger.ac.uk/) - A Hub for Preclinical Cancer Models - Annotation, Genomics & Functional Datasets. Over 1900 cancer cell line and organoid models catalogued.
- [sandbox.bio](https://sandbox.bio/) - Learn how to use bioinformatics tools right from your browser. Everything runs in a sandbox, so you can experiment all you want.
- [Gosling](http://gosling-lang.org/) - A Grammar-based Toolkit for Scalable and Interactive Genomics Data Visualization.

### Bioinformatics Databases

 - [BioGRID](https://thebiogrid.org/) - Database of Protein, Chemical, and Genetic Interactions
 - [Intact](https://www.ebi.ac.uk/intact/) - IntAct provides a freely available, open source database system and analysis tools for molecular interaction data. All interactions are derived from literature curation or direct user submissions and are freely available.
 - [I2D - Interologous Interaction Database](http://ophid.utoronto.ca/ophidv2.204/) -  To faciliate experimentation and integrated computational analysis with model organism PPI networks. I2D is developed and maintained by Jurisica Lab at Ontario Cancer Institute, PMH.
 - [KEGG](https://www.genome.jp/kegg/) - KEGG is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies.
 - [Lnc2Catlas](https://lnc2catlas.bioinfotech.org/) - An atlas of long noncoding RNAs associated with risk to cancer. They provide interactions between 33 different cancers and 27,670 LncRNA transcripts. Three methods were used to explore the relationship between cancers and LncRNAs, via SNPs, proteins and FPKMs separately. 
 - [LNCipedia](https://hg38.lncipedia.org/) - A comprehensive compendium of human long non-coding RNAs.
 - [lncRInter-lncRNA Interaction](http://bioinfo.life.hust.edu.cn/lncRInter/) - lncRInter is reliable and high quality lncRNA interaction database containing experimentally validated data.
 - [NONCODE](http://www.noncode.org/) - NONCODE is an integrated knowledge database dedicated to non-coding RNAs.
 - [NPInter](http://www.bioinfo.org/NPInter/)
 - [Reactome Pathway Database](http://www.reactome.org/) - Reactome is a free, open-source, curated and peer-reviewed pathway database. Their goal is to provide intuitive bioinformatics tools for the visualization, interpretation and analysis of pathway knowledge to support basic research, genome analysis, modeling, systems biology and education.
 - [RNAInter](https://www.rna-society.org/rnainter/) - RNA Interactome Database.
 - [starBase v2.0](http://starbase.sysu.edu.cn/)
 - [STRING](https://string-db.org/) - Functional protein association networks.
 - [EMBL-EBI Expression Atlas](https://www.ebi.ac.uk/gxa/home) - Expression Atlas is an open science resource that gives users a powerful way to find information about gene and protein expression. Our mission is to provide the scientific community with freely available information on the abundance and localisation of RNA (and proteins) across species and biological conditions such as different tissues, cell types, developmental stages and diseases among others.
 - [LncExpDB: Home](https://ngdc.cncb.ac.cn/lncexpdb/)
 
 ### General Resources

 - [An Introduction to Applied Bioinformatics](http://readiab.org/) - An Introduction to Applied Bioinformatics (or IAB) is a free, open source interactive text that introduces readers to core concepts of bioinformatics in the context of their implementation and application. IAB is primarily being developed by Greg Caporaso (GitHub/Twitter: [@gregcaporaso](https://github.com/gregcaporaso)) in the [Caporaso Lab](http://www.caporasolab.us/) at Northern Arizona University.
 - [Karyoploter Tutorial](https://bernatgel.github.io/karyoploter_tutorial/) - A tutorial and a set of examples on using the R/Bioconductor package [karyoploteR](http://bioconductor.org/packages/karyoploteR) to plot whole genomes with arbitrary data on them.
 - [PH525x series - Biomedical Data Science](http://genomicsclass.github.io/book/) - This book covers several of the statistical concepts and data analytic skills needed to succeed in data-driven life science research. They go from relatively basic concepts related to computing p-values to advanced topics related to analyzing high-throughput data.
 - [QC Fail Sequencing](https://sequencing.qcfail.com/) - Articles about common next-generation sequencing problems.
 - [Teaching Materials](http://www.langmead-lab.org/teaching-materials/) - Langmead teaching materials.
 - [Analyzing RNA-seq data with DESeq2](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html) - This vignette explains the use of the DESeq2 package and demonstrates typical workflows.
 - [An RNA-seq workflow](http://www.bioconductor.org/help/workflows/rnaseqGene/) on the Bioconductor website covers similar material to this vignette but at a slower pace, including the generation of count matrices from FASTQ files.
 - [crazyhottommy/RNA-seq-analysis](https://github.com/crazyhottommy/RNA-seq-analysis) - RNAseq analysis notes from Tommy Tang.  A comprehensive list with various resources.
 - [Finding and quantifying new transcripts](https://galaxyproject.org/tutorials/nt_rnaseq/) - A Galaxy tutorial on comparing RNAseq data from two distinct cell types in order to identify and quantify differentially expressed transcripts.
 - [Generating an using Ensembl based annotation packages](https://www.bioconductor.org/packages/release/bioc/vignettes/ensembldb/inst/doc/ensembldb.html) - A tutorial on the ensembldb Bioconductor package. The ensembldb package provides functions to create and use transcript centric annotation databases/packages. The functionality and data is similar to that of the TxDb packages from the GenomicFeatures package, but, in addition to retrieve all gene/transcript models and annotations from the database, the ensembldb package provides also a filter framework allowing to retrieve annotations for specific entries like genes encoded on a chromosome region or transcript models of lincRNA genes.
 - [Bulk RNA-seq Data Standards and Processing Pipeline - ENCODE](https://www.encodeproject.org/data-standards/rna-seq/long-rnas/) - The ENCODE Bulk RNA-seq pipeline can be used for both replicated and unreplicated, paired-ended or single-ended, and strand-specific or non-strand specific RNA-seq libraries. Libraries must be generated from mRNA (poly(A)+, rRNA-depleted total RNA, or poly(A)- populations that are size-selected to be longer than approximately 200 bp.
 - [RNA-seq workflow: gene-level exploratory analysis and differential expression](https://bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html) - A walk through an end-to-end gene-level RNA-seq differential expression workflow using Bioconductor packages.
 - [RNAseq - an introduction](https://galaxyproject.org/tutorials/rb_rnaseq/) - An introductory tutorial on RNA-seq by the Galaxy team.
 - [RNA-seqlopedia](http://rnaseq.uoregon.edu/)
 - [Data Analysis in Genome Biology](https://girke.bioinformatics.ucr.edu/GEN242/index.html) - GEN242 is a graduate class taught at the University of California, Riverside. Currently, the main instructor is Thomas Girke. This course introduces algorithms, statistical methods and data analysis programming routines relevant for genome biology. It consists of three main components: lectures, hands-on practicals and student course projects. The lecture topics cover databases, sequence (NGS) analysis, phylogenetics, comparative genomics, genome-wide profiling methods, network biology and more.
 - [Weighted gene co-expression network analysis with TCGA RNAseq data](https://bioconductor.riken.jp/packages/3.4/bioc/vignettes/CVE/inst/doc/WGCNA_from_TCGA_RNAseq.html)
 - [Introduction to RNA-Seq using high-performance computing](https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon-flipped/schedule/links-to-lessons.html)
 - [Network analysis with R and igraph: NetSci X Tutorial - Katya Ognyanova](https://kateto.net/networks-r-igraph) - igraph R tutorial.
 - [Homepage of Dave Tang](https://davetang.org/)
 - [crazyhottommy/getting-started-with-genomics-tools-and-resources](https://github.com/crazyhottommy/getting-started-with-genomics-tools-and-resources)
 - [rafalab - HarvardX Biomedical Data Science Open Online Training](https://rafalab.github.io/pages/harvardx.html)

### Programs

 - [catRAPID](http://s.tartaglialab.com/page/catrapid_group) - catRAPID is an algorithm to estimate the binding propensity of protein-RNA pairs. By combining secondary structure, hydrogen bonding and van der Waals contributions, catRAPID predicts protein-RNA associations with great accuracy.
 - [FASTX-Toolkit](http://hannonlab.cshl.edu/fastx_toolkit/) - The FASTX-Toolkit is a collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing.
 - [FeatureCounts](http://bioinf.wehi.edu.au/featureCounts/) - Part of [subread](http://subread.sourceforge.net/) package.
 - [lncPro](http://bioinfo.bjmu.edu.cn/lncpro/)
 - [PIVOT](http://kim.bio.upenn.edu/software/pivot.shtml) - A Platform for Interactive analysis and Visualization Of Transcriptomics data. Its goal is to facilitate fast and interactive RNA-Seq data analysis and visualization. Current version of PIVOT supports routine RNA-Seq data analysis including normalization, differential expression analysis, dimension reduction, correlation analysis, clustering and classification. Users can complete workflows of DESeq2, monocle and scde package with just a few button clicks. All analysis reports can be exported, and the program state can be saved, loaded and shared. 
 - [MultiQC](http://multiqc.info/) - MultiQC searches a given directory for analysis logs and compiles a HTML report. It's a general use tool, perfect for summarising the output from numerous bioinformatics tools.
 - [PCR Primer Stats](http://www.bioinformatics.org/sms2/pcr_primer_stats.html) - PCR Primer Stats accepts a list of PCR primer sequences and returns a report describing the properties of each primer, including melting temperature, percent GC content, and PCR suitability. Use PCR Primer Stats to evaluate potential PCR primers.
 - [Qualimap](http://qualimap.bioinfo.cipf.es/) - Evaluating next generation sequencing alignment data. Qualimap 2 is a platform-independent application written in Java and R that provides both a Graphical User Inteface (GUI) and a command-line interface to facilitate the quality control of alignment sequencing data and its derivatives like feature counts.
 - [RSeQC](http://rseqc.sourceforge.net/) - An RNA-seq Quality Control Package. RSeQC package provides a number of useful modules that can comprehensively evaluate high throughput sequence data especially RNA-seq data. Some basic modules quickly inspect sequence quality, nucleotide composition bias, PCR bias and GC bias, while RNA-seq specific modules evaluate sequencing saturation, mapped reads distribution, coverage uniformity, strand specificity, transcript level RNA integrity etc.
 - https://github.com/sestaton/Pairfq
 - http://mafft.cbrc.jp/alignment/server/
 - http://trinotate.github.io/
 - [LncRNA-RNA interaction](http://www.cuilab.cn/lnctar) - LncTar: an efficient tool for predicting RNA targets of lncRNAs.
 
 ## R

 - [A Compendium of Clean Graphs in R](http://shinyapps.org/apps/RGraphCompendium/index.php) - This compendium facilitates the creation of good graphs by presenting a set of concrete examples, ranging from the trivial to the advanced. The graphs can all be reproduced and adjusted by copy-pasting code into the R console. The majority of the plots have been created in base R, but you will encounter some examples in ggplot.
 - [R for Data Science](https://r4ds.had.co.nz/) - This book will teach you how to do data science with R: You’ll learn how to get your data into R, get it into the most useful structure, transform it, visualise it and model it.
 - [The R Graph Gallery – Help and inspiration for R charts](https://www.r-graph-gallery.com/) - A collection of charts made with the R programming language. Hundreds of charts are displayed in several sections, always with their reproducible code available.
 - [R Graphics Cookbook, 2nd edition](https://r-graphics.org/) - A practical guide that provides more than 150 recipes to help you generate high-quality graphs quickly, without having to comb through all the details of R’s graphing systems.
 - [R Packages](https://r-pkgs.org/) -  In this book you’ll learn how to turn your code into packages that others can easily download and use.
 - [The Bioconductor 2018 Workshop Compilation](https://bioconductor.github.io/BiocWorkshops/index.html) - Workshop materials are available as a book in html, pdf, and eBook format at https://bioconductor.github.io/BiocWorkshops/. Workshops are organized by level and topic according to numbers, as described below. Every workshop starts with a syllabus that will help you to decide whether it matches your learning goals.
 - [What They Forgot to Teach You About R](https://rstats.wtf/) - 
 - [Descriptive statistics in R](https://statsandr.com/blog/descriptive-statistics-in-r/) - Useful functions.

### R Packages

 - [containerit](https://github.com/o2r-project/containerit) - containerit packages R script/session/workspace and all dependencies as a Docker container by automagically generating a suitable Dockerfile. The package’s website is https://o2r.info/containerit/.

## Books

 - [Computational Genomics with R](https://compgenomr.github.io/book/)
 - [Introduction to data science](https://rafalab.dfci.harvard.edu/dslivro/)
 - [Introduction to Bioinformatics and Computational Biology](https://liulab-dfci.github.io/bioinfo-combio/)
 - [R Markdown: The Definitive Guide](https://bookdown.org/yihui/rmarkdown/)
 - [Introduction - An Introduction to Snakemake for Bioinformatics](https://farm.cse.ucdavis.edu/~ctbrown/2023-snakemake-book-draft/)

## Web Apps

 - [ARCHS4](https://maayanlab.cloud/archs4/index.html) - ARCHS4 provides access to gene counts from HiSeq 2000, HiSeq 2500 and NextSeq 500 platforms for human and mouse experiments from GEO and SRA.
 - [BoxplotR](http://shiny.chemgrid.org/boxplotr/) - A web-tool for generation of box plots.
 - [g:Profiler](https://biit.cs.ut.ee/gprofiler/gost) - g:Profiler performs functional enrichment analysis, also known as over-representation analysis (ORA) or gene set enrichment analysis, on input gene list.
 - [Venny 2.1.0](http://bioinfogp.cnb.csic.es/tools/venny/) - Venn diagrams made easy.
 - [BioJupies](https://amp.pharm.mssm.edu/biojupies/) - With BioJupies you can produce in seconds a customized, reusable, and interactive report from your own raw or processed RNA-seq data through a simple user interface.
 
## Courses

- [Modern Statistics for Modern Biology](https://www.huber.embl.de/msmb/index.html) - The aim of this book is to enable scientists working in biological research to quickly learn many of the important ideas and methods that they need to make the best of their experiments and of other available data.
- [Welcome to Chromebook Data Science](http://jhudatascience.org/chromebookdatascience/index.html) - Chromebook Data Science (CBDS) is a free, massive open online educational program offered through Leanpub to help anyone who can read, write, and use a computer to move into data science.
- [quinlan-lab/applied-computational-genomics](https://github.com/quinlan-lab/applied-computational-genomics) - Applied Computational Genomics Course at UU: Spring 2020 by Aaron Quinlan.
- [The Docker Handbook - 2021 Edition](https://www.freecodecamp.org/news/the-docker-handbook/)
- [Algotithms Full Course - Youtube](https://www.youtube.com/watch?v=D6Q_wHrzxDs) - +11 hours.
- [Linear Algebra Full College Course - Youtube](https://www.youtube.com/watch?v=JnTa9XtvmfI) - +11 hours.
- [Big O Notation](https://www.youtube.com/watch?v=Mo4vesaut8g) - 2 hours.
- [Seeing Theory](https://seeing-theory.brown.edu/#firstPage) - A visual introduction to probability and statistics.
- [STAT555](https://online.stat.psu.edu/stat555/node/4/) - Basic Statistical Inference for Bioinformatics Studies
